jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHolder.java

```diff
@@ -294,6 +294,22 @@ public void setForcedPath(String forcedPath)
         _forcedPath = forcedPath;
     }
 
+    private void copyClassServlet(ServletHolder holder) throws ServletException
+    {
+        if (holder.getClassName() != null)
+            this.setClassName(holder.getClassName());
+        // TODO adding this after setting the classname (or instead of) fixes the exceptions.
+        /*
+        Class<? extends Servlet> servletClass = holder.getHeldClass();
+        if (servletClass != null)
+            this.setHeldClass(servletClass);
+        */
+        if (holder.getServlet() != null)
```

Comment:
I'm not sure this is correct. I don't think we have any current behaviour where we reuse an _instance_ of a servlet across ServletHolders.